doc/sphinx/source/users-guide/base/forloops.rst

```diff
@@ -0,0 +1,113 @@
+:title: Users Guide: For loops
+
+.. _ug-forloops:
+
+For-loops
+=========
+
+*For-loops* are Chapel's means of expressing serial iteration as
+driven by an iterator function or iterable expression.  As a simple
+example, the following for-loop loops over the integers 1 through 5,
+printing a message per iteration:
+
+.. literalinclude:: examples/users-guide/base/forloops.chpl
+  :caption:
+  :language: chapel
+  :lines: 3-4
+
+More specifically, this for-loop invokes the default iterator for the
+*range* expression ``1..5``, storing the values that it yields in a
+per-iteration variable named *i*.  The body of the loop then refers to
+*i* as a means of customizing the messages that are printed.  The
+resulting output is:
+
+.. literalinclude:: examples/users-guide/base/forloops.good
+  :language: text
+  :lines: 1-5
+
+For-loop Index Variables
+------------------------
+
+Note that for-loop index variables, like *i* above, are new symbols
+that are defined by the for-loop and only available within its body.
+Thus, the following code actually declares two variables named *i*
+where the for-loop's index variable temporarily shadows the original
```

Comment:
I'm trying to think why shadows is an advanced concept to me, and I think it's just because I learned it late. This is probably fine, just doing some metacognition.